File: apps/web/scripts/sync-glossary-quiz.mjs
/**
 * Copy API healthcare glossary → web quiz pool (runs automatically via prebuild/predev).
 * Source of truth: apps/api/app/data/healthcare_vocabulary.json
 */
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const webRoot = path.join(__dirname, "..");
const repoRoot = path.join(webRoot, "../..");

const SRC = path.join(repoRoot, "apps/api/app/data/healthcare_vocabulary.json");
const DEST = path.join(
  webRoot,
  "src/content/assessment/data/healthcare_vocabulary.json",
);

function main() {
  if (!fs.existsSync(SRC)) {
    console.error(`sync-glossary-quiz: missing source ${SRC}`);
    process.exit(1);
  }

  const raw = fs.readFileSync(SRC, "utf8");
  const data = JSON.parse(raw);
  const count = Object.keys(data).length;

  fs.mkdirSync(path.dirname(DEST), { recursive: true });

  const unchanged =
    fs.existsSync(DEST) && fs.readFileSync(DEST, "utf8") === `${JSON.stringify(data, null, 2)}\n`;

  if (!unchanged) {
    fs.writeFileSync(DEST, `${JSON.stringify(data, null, 2)}\n`);
    console.log(`sync-glossary-quiz: wrote ${count} entries → ${path.relative(webRoot, DEST)}`);
  } else {
    console.log(`sync-glossary-quiz: up to date (${count} entries)`);
  }
}

main();

File: scripts/sync-glossary-quiz.py
#!/usr/bin/env python3
"""Copy API healthcare glossary into web quiz pool. Prefer: pnpm sync:glossary (Node, runs on prebuild)."""
from __future__ import annotations

import json
import shutil
import subprocess
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
NODE_SCRIPT = ROOT / "apps" / "web" / "scripts" / "sync-glossary-quiz.mjs"


def main() -> None:
    if NODE_SCRIPT.is_file():
        subprocess.run(["node", str(NODE_SCRIPT)], cwd=ROOT / "apps" / "web", check=True)
        return

    src = ROOT / "apps" / "api" / "app" / "data" / "healthcare_vocabulary.json"
    dest = ROOT / "apps" / "web" / "src" / "content" / "assessment" / "data" / "healthcare_vocabulary.json"
    if not src.is_file():
        raise SystemExit(f"Missing source glossary: {src}")
    dest.parent.mkdir(parents=True, exist_ok=True)
    shutil.copy2(src, dest)
    data = json.loads(dest.read_text(encoding="utf-8"))
    print(f"Synced {len(data)} glossary entries to {dest.relative_to(ROOT)}")


if __name__ == "__main__":
    main()
